core/server/apps/amp/lib/helpers/amp_content.js

```diff
@@ -4,15 +4,43 @@
 // Turns content html into a safestring so that the user doesn't have to
 // escape it or tell handlebars to leave it alone with a triple-brace.
 //
-// Converts normal HTML into AMP HTML.
+// Converts normal HTML into AMP HTML with Amperize module and uses a cache to return it from
+// there if available. The cacheId is a combination of `updated_at` and the `slug`.
 
 var hbs             = require('express-hbs'),
-    ampContent;
+    Promise         = require('bluebird'),
+    Amperize        = require('amperize'),
+    moment          = require('moment'),
+    amperizeCache = {};
 
-ampContent = function () {
-    // var root = options.data.root;
-    // console.log('root:', root);
-    return new hbs.handlebars.SafeString(this.html);
-};
+function getAmperizeHTML(html, cacheId) {
+    if (!amperizeCache[cacheId]) {
+        return new Promise(function (resolve, reject) {
+            new Amperize().parse(html, function (err, res) {
+                if (err) {
+                    return reject(err);
+                }
+
+                amperizeCache[cacheId] = res;
+                return resolve(amperizeCache[cacheId]);
+            });
+        });
+    }
+
+    return Promise.resolve(amperizeCache[cacheId]);
+}
+
+function ampContent() {
+    var cacheId = moment(new Date(this.updated_at)) + this.slug,
```

Comment:
suggestion: the in process cache will grow over time, because we always keep the old cache entries.
Plus a post slug can change, a post.id cannot. So if we take the post.id as key for the `amperizeCache`, we would always override the cache entry for a post if the post has changed. (`amperizeCache[post.id] = cacheEntry`).